### ISSUE TITLE
Bug: `.catch` causes `undefined` types inside `z.object`

### ISSUE THREAD

**Issue by jlalmes** [colinhacks/zod#1807]
### Bug: `.catch` causes `undefined` types inside `z.object`.

This issue appears to be different to https://github.com/colinhacks/zod/issues/1737 (fixed by PR https://github.com/colinhacks/zod/pull/1733)

🚀 Example outside `z.object`:
```ts
import { z } from 'zod';

const schema = z.string().catch('asdf');

const result = schema.parse(undefined);
//      ^? string

console.log(result); // "asdf"
```

🚧 Example inside `z.object`:
```ts
import { z } from 'zod';

const schema = z.object({
  value: z.string().catch('asdf'),
});

const result = schema.parse({});
//      ^? { value?: string }

console.log(result); // { value: "asdf" }

// Why is `result.value` possibly undefined?
```

Reproduction:
https://stackblitz.com/edit/typescript-sucpqa?devToolsHeight=100&file=index.ts

**Comment by jlalmes** [comment]
Whoops, just taken a second look, it seems this was already resolved by the merged PR https://github.com/colinhacks/zod/pull/1733. Any timeline for when this fix might be published @colinhacks to npm?

**Comment by colinhacks** [comment]
This has landed in `zod@3.20.3`, sorry for the long delay.
